### ISSUE TITLE
MatchSpec parses incorrectly when channel is passes as a URL

### ISSUE THREAD

**Issue by travishathaway** [conda/rattler#793]
### What happened?

While using the `py-rattler` library, I discovered a possible problem with how MatchSpecs are parsed when passing the channel name as a URL

```python
>>> from rattler import MatchSpec
>>> m = MatchSpec("https://conda.anaconda.org/conda-forge::python[version=3.9]")
>>> m
MatchSpec("*")
>>> print(m.channel)
None
```

The following happens when specifying the `channel` as a MatchSpec key:

```python
>>> from rattler import MatchSpec
>>> m = MatchSpec("python[channel=https://conda.anaconda.org/conda-forge,version=3.9]")
InvalidMatchSpecException                 Traceback (most recent call last)
Cell In[21], line 1
----> 1 m = MatchSpec("python[channel=https://conda.anaconda.org/python/conda,version=3.9]")

File ~/opt/testminiconda/envs/pyrattler/lib/python3.12/site-packages/rattler/match_spec/match_spec.py:85, in MatchSpec.__init__(self, spec, strict)
     83 def __init__(self, spec: str, strict: bool = False) -> None:
     84     if isinstance(spec, str):
---> 85         self._match_spec = PyMatchSpec(spec, strict)
     86     else:
     87         raise TypeError(
     88             "MatchSpec constructor received unsupported type" f" {type(spec).__name__!r} for the 'spec' parameter"
     89         )

InvalidMatchSpecException: invalid bracket key: channel
```

### What should happen?

I would expect rattler to be able to parse full URLs properly. This is currently also a bug in conda as reported here:

- https://github.com/conda/conda/issues/14074

But, micromamba appears to be parsing this correctly.

**Comment by ruben-arts** [comment]
Just to verify. The channel name of `https://software.repos.intel.com/python/conda` should be `conda` right? Not `python/conda`.

**Comment by baszalmstra** [comment]
We just merged #792 which fixes:

```python
>>> from rattler import MatchSpec
>>> m = MatchSpec("https://conda.anaconda.org/conda-forge::python[version=3.9]")
>>> m
MatchSpec("conda-forge::python ==3.9")
>>> m.channel
Channel(name="conda-forge", base_url="https://conda.anaconda.org/conda-forge/")
```

Parsing `python[channel=https://conda.anaconda.org/python/conda,version=3.9]`  still fails because we dont recognize the `channel` key yet. That will be next.
